src/sentry/static/sentry/app/views/performance/transactionSummary/header.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import {Location} from 'history';
+
+import {Organization, Project} from 'app/types';
+import EventView from 'app/utils/discover/eventView';
+import Feature from 'app/components/acl/feature';
+import CreateAlertButton from 'app/components/createAlertButton';
+import * as Layout from 'app/components/layouts/thirds';
+import ButtonBar from 'app/components/buttonBar';
+import {trackAnalyticsEvent} from 'app/utils/analytics';
+import Breadcrumb from 'app/views/performance/breadcrumb';
+
+import KeyTransactionButton from './keyTransactionButton';
+
+type Props = {
+  eventView: EventView;
+  location: Location;
+  organization: Organization;
+  projects: Project[];
+  transactionName: string;
+};
+
+class TransactionHeader extends React.Component<Props> {
+  trackAlertClick(errors?: Record<string, boolean>) {
+    const {organization} = this.props;
+    trackAnalyticsEvent({
+      eventKey: 'performance_views.summary.create_alert_clicked',
+      eventName: 'Performance Views: Create alert clicked',
+      organization_id: organization.id,
+      status: errors ? 'error' : 'success',
+      errors,
+      url: window.location.href,
+    });
+  }
+
+  handleIncompatibleQuery: React.ComponentProps<
+    typeof CreateAlertButton
+  >['onIncompatibleQuery'] = (incompatibleAlertNoticeFn, errors) => {
+    this.trackAlertClick(errors);
+    const incompatibleAlertNotice = incompatibleAlertNoticeFn(() =>
+      this.setState({incompatibleAlertNotice: null})
+    );
+    this.setState({incompatibleAlertNotice});
```

Comment:
This component doesn't have any state defined in it. Should this be signaling up to the containing component instead?

Reply:
ah my bad